test/e2e/cross_os/test_job_submissions.py

```diff
@@ -632,47 +632,207 @@ def test_worker_uses_job_attachment_configuration(
+                    }
+                )
             )
-            job_output_downloader.download_job_output()
-
-            with (
-                open(os.path.join(job_bundle_path, "files", "test_input_file"), "r") as input_file,
-                open(
-                    os.path.join(
-                        tmp_dir_name,
-                        "output_file",
-                    ),
-                    "r",
-                    encoding="utf-8-sig",
-                ) as output_file,
-            ):
-                input_file_content: str = input_file.read()
-                output_file_content = output_file.read()
-
-                # Verify that the output file content is the input file content plus the uuid we appended in the job
-                assert output_file_content == (input_file_content + test_run_uuid)
+
+        config = configparser.ConfigParser()
+
+        set_setting("defaults.farm_id", deadline_resources.farm.id, config)
+        set_setting("defaults.queue_id", deadline_resources.queue_a.id, config)
+
+        job_id: Optional[str] = api.create_job_from_job_bundle(
+            job_bundle_path,
+            job_parameters,
+            priority=99,
+            max_retries_per_task=0,
+            config=config,
+            queue_parameter_definitions=[],
+        )
+        assert job_id is not None
+
+        job_details = Job.get_job_details(
+            client=deadline_client,
+            farm=deadline_resources.farm,
+            queue=deadline_resources.queue_a,
+            job_id=job_id,
+        )
+        job = Job(
+            farm=deadline_resources.farm,
+            queue=deadline_resources.queue_a,
+            template={},
+            **job_details,
+        )
+
+        # Query the session to check for progress percentage
+        complete_percentage: float = 0
+
+        @backoff.on_predicate(
+            wait_gen=backoff.constant,
+            max_time=30,
+            interval=2,
+        )
+        def check_percentage(complete_percentage):
+            sessions = deadline_client.list_sessions(
+                farmId=job.farm.id, queueId=job.queue.id, jobId=job.id
+            ).get("sessions")
+
+            if sessions:
+                session_actions = deadline_client.list_session_actions(
+                    farmId=job.farm.id,
+                    queueId=job.queue.id,
+                    jobId=job.id,
+                    sessionId=sessions[0]["sessionId"],
+                ).get("sessionActions")
+
+                for session_action in session_actions:
+                    if "syncInputJobAttachments" in session_action["definition"]:
+                        assert complete_percentage <= session_action["progressPercent"]
+                        complete_percentage = session_action["progressPercent"]
+                        return complete_percentage == 100
+
+        assert check_percentage(complete_percentage)
+
+        shutil.rmtree(job_bundle_path)
```

Comment:
This is not safe, this code may never get ran due to a test failure before this line, and then the job bundle path will never get deleted. i think we should either use a temporary directory like https://docs.python.org/3/library/tempfile.html or use a `finally` clause to clean up the directory.